os/net/netdev/netdev_ioctl.c

```diff
@@ -1003,6 +1006,58 @@ static int netdev_rtioctl(FAR struct socket *sock, int cmd, FAR struct rtentry *
 }
 #endif							/* CONFIG_NET_ROUTE */
 
+#if CONFIG_NET_LWIP
+/****************************************************************************
+ * Function: lwip_func_ioctl
+ *
+ * Description:
+ *   Call lwip API
+ *
+ * Parameters:
+ *   arg Type of the information to get
+ *
+ * Returned Value:
+ *   0 on success, an error code on failure
+ *
+ ****************************************************************************/
+
+int lwip_func_ioctl(void *arg)
```

Comment:
ioctl calls set_errno based on return value of each sub-ioctl(In your case this function)
So user will check return value by calling errno macro, and will parse error value in errno.h
But most of return value of functions here defined in netdb.h, you have to find another way.
See below
```
   fd = socket();
   ret = ioctl(fd, IOC_XXX, &arg);
   if (ret < 0) {
        dbg("error, ret : %d\n", errno); 
   }
```
Assume that errno is EAI_NONAME, but there's no return value 'EAI_NONAME' in errno.h.
So below is one option
```
int lwip_func_ioctl(void *arg)
...
case GETADDRINFO:
     ...
     ret = lwip_getaddrinfo();
     if (ret == EAI_NONAME) {
         dbg("errnor, ret : %d\n", ret);
         return EMFILE;
     }
```
It is common ioctl usage.